From a189f3863881adc3025662cac00bde1c778bcedf Mon Sep 17 00:00:00 2001
From: Zach Reizner <zachr@google.com>
Date: Mon, 6 Apr 2015 11:25:12 -0700
Subject: [PATCH 04/17] CHROMIUM: vgem: add DRM_IOCTL_VGEM_MODE_MAP_DUMB
 support

This is the libdrm counterpart of the kernel patch in CL:263991:
DRM_IOCTL_MODE_MAP_DUMB is not allowable for use by render nodes, so we
add DRM_IOCTL_VGEM_MODE_MAP_DUMB, which behaves exactly the same as
DRM_IOCTL_MODE_MAP_DUMB, and is allowed on render nodes.

TEST=the file /usr/include/libdrm/vgem_drm.h under the board's build
BUG=chromium:469666
BUG=b:26575005

Signed-off-by: Zach Reizner <zachr@google.com>
Change-Id: I539b15b47c08a92ca50b21a2b370ec44cdfae586
Signed-off-by: Randy Li <randy.li@rock-chips.com>
---
 Makefile.sources       |  1 +
 include/drm/vgem_drm.h | 33 +++++++++++++++++++++++++++++++++
 2 files changed, 34 insertions(+)
 create mode 100644 include/drm/vgem_drm.h

diff --git a/Makefile.sources b/Makefile.sources
index 1a1f0fe..03c44f0 100644
--- a/Makefile.sources
+++ b/Makefile.sources
@@ -33,6 +33,7 @@ LIBDRM_INCLUDE_H_FILES := \
 	include/drm/sis_drm.h \
 	include/drm/tegra_drm.h \
 	include/drm/vc4_drm.h \
+	include/drm/vgem_drm.h \
 	include/drm/via_drm.h
 
 LIBDRM_INCLUDE_VMWGFX_H_FILES := \
diff --git a/include/drm/vgem_drm.h b/include/drm/vgem_drm.h
new file mode 100644
index 0000000..9be928f
--- /dev/null
+++ b/include/drm/vgem_drm.h
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015 The Chromium OS Authors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER(S) OR AUTHOR(S) BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef _VGEM_DRM_H_
+#define _VGEM_DRM_H_
+
+#include "drm.h"
+
+#define DRM_VGEM_MODE_MAP_DUMB 0x00
+
+#define DRM_IOCTL_VGEM_MODE_MAP_DUMB \
+	DRM_IOWR(DRM_COMMAND_BASE + DRM_VGEM_MODE_MAP_DUMB, struct drm_mode_map_dumb)
+
+#endif
-- 
2.7.4

